### DISCUSSION TITLE
Image not rendering but renders locally?

### DISCUSSION THREAD

**Discussion by RobynLJackson** [community/community#22777] [Other Feature Feedback, Questions, & Ideas]
<p>Hello! I’m Robyn and I’m a student of web development and Git is something I am working hard to learn/get proficient with. I just uploaded a new repo (simple advice page). My CSS wasn’t working, but I seem to have fixed that although I still don’t know why it wasn’t showing up. My file structure and file paths in my html are all working fine locally, but were not on gitpages. I cannot get my image to load on my published page. Can someone help me with this please? Here’s the link: <a href="https://robynljackson.github.io/robyns-advice/robyn-j-advice.html" rel="noopener nofollow ugc">https://robynljackson.github.io/robyns-advice/robyn-j-advice.html</a></p>
<p>**Correction: I THOUGHT I fixed the css issue. It’s not working either.</p>
<p>Thank you!</p>

**Comment by otter-computer** [comment]
<p>Hey <a class="mention" href="/robynljackson">@RobynLJackson</a>, welcome to the GitHub Support Community!</p>
<p>It looks like you’ve already managed to fix this by updating your asset links to use relative links to your repository instead of using absolute, local links to your files on disk.</p>
<p>Feel free to drop another message or open a new thread if you need help with anything else!</p>

**Comment by RobynLJackson** [comment]
<p>Hello <a class="mention" href="/thomasshaped">@thomasshaped</a>! I appreciate your time and help. I’m glad to have the links update confirmed, however, the page is not displaying (a 404 error). I’m wondering at this point if it’s because there is an index.html and an advice.html page (the 2nd page which should open with the link from the index.html)? I still think it’s a file structure issue, but can’t figure out what the solution is yet. Can you help please? Here’s the link to the advice.html that should come up on gitpages: [<a href="https://robynljackson.github.io/robyns-advice/advice/robyn-j-advice.html" rel="noopener nofollow ugc">https://robynljackson.github.io/robyns-advice/advice/robyn-j-advice.html</a>]. I’m working hard to learn git and become proficient at it. Hosting this simple site should be something I know how to do without a problem. I appreciate any help you can provide!<br>
Also…here is the local directory:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/1/11e5719ee1b7b0b69bd1374ae6500d4f1238d130.png" data-download-href="/uploads/short-url/2yjHhDIIRsEgxomjWf5b73agrQI.png?dl=1" title="robyns-advice-local-directory" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/49930307/181073114-71b60903-5817-442c-a2fc-a7f12f81c9d0.png" alt="robyns-advice-local-directory" data-base62-sha1="2yjHhDIIRsEgxomjWf5b73agrQI" width="690" height="473" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/1/11e5719ee1b7b0b69bd1374ae6500d4f1238d130_2_690x473.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/1/11e5719ee1b7b0b69bd1374ae6500d4f1238d130_2_1035x709.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/1/11e5719ee1b7b0b69bd1374ae6500d4f1238d130.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/1/11e5719ee1b7b0b69bd1374ae6500d4f1238d130_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">robyns-advice-local-directory</span><span class="informations">1176×807 63.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><p></p>

**Comment by otter-computer** [comment]
<p>It looks like you’ve got an extra <code>/advice/</code> in that URL so you’ll need to update the link in your <code>index.html</code> file to remove it. That page is online here:</p>
<p><code>https://robynljackson.github.io/robyns-advice/robyn-j-advice.html</code><br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <a href="https://robynljackson.github.io/robyns-advice/robyn-j-advice.html" target="_blank" rel="noopener">robynljackson.github.io</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://robynljackson.github.io/robyns-advice/robyn-j-advice.html" target="_blank" rel="noopener">Robyn's Advice For Newbies</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Alternatively, if you <em>want</em> the <code>/advice/</code> section in your URL, you’ll need to create a folder called <code>advice</code> and move the <code>robyn-j-advice.html</code> file inside it.</p>

**Comment by RobynLJackson** [comment]
<p>Greetings <a class="mention" href="/thomasshaped">@thomasshaped</a>. Thank you for leading me through my tangle of file path and file structure errors.</p>
<p>I updated the link as you pointed out. Changes have not updated (or worked) yet.  I see/get that the file structure of the directory is sloppy and that’s a problem with hierarchy and page loading.</p>
<p>Thank you for clarifying the issues for me!</p>

**Comment by RobynLJackson** [comment]
<p>Hi again <a class="mention" href="/thomasshaped">@thomasshaped</a>.</p>
<p>At the command line I rm the index.html, renamed the advice.html to index.html. My local directory and the repo is up to date.</p>
<p>I pushed these changes to “git push origin gh-pages”. Is there something else I need to do to push these changes to my hosted page? How do I delete the index.html file in the main branch of my repo so I can merge the two branches?</p>
<p>Thank you!</p>

**Comment by RobynLJackson** [comment, marked as answer]
<p><a class="mention" href="/thomasshaped">@thomasshaped</a><br>
I resolved it and my git pages link works perfectly. I made all the changes in VSC and the command line.</p>
<p>Thank you for your time and serious patience!<br>
[robyns-advice]<br>
(<a href="https://robynljackson.github.io/robyns-advice/" rel="noopener nofollow ugc">https://robynljackson.github.io/robyns-advice/</a>)</p>
